### ISSUE TITLE
keep-alive connections and http / https agents

### ISSUE THREAD

**Issue by dendril** [bitinn/node-fetch#110]
Hey!

I have been dealing with some performance stuff using node-fetch, and after a lot of debugging I finally found the issue.

Use case: around 10k requests to the same cdn using keep-alive.

```js
const fetch = require('node-fetch');
const agent = new https.Agent({
	keepAlive: true,
	keepAliveMsecs: 1500,
	maxSockets: 70
});

for (let i=0; i < 10000; i++) fetch(`https://cdn.test.io/${i}`, {
	headers,
	agent,
	compress: true
});
```

If you check https://github.com/bitinn/node-fetch/blob/master/index.js#L77-L79 we are explicitly saying `Connection: close`, so, each time I made a request to the server, I need to negotiate a new connection before make a request, doesn't matter if I could reuse an already established connection from the pool.

We should change the default to do nothing, because if I set keepAlive in the agent, `Connection: close` prevents the expected behavior. 

What do you think?

**Comment by bitinn** [comment]
If you pass `keep-alive` to `connection` header does it work?

**Comment by dendril** [comment]
Yeap, but the API hide this implementation detail. Nowadays in http servers `connection: keep-alive` is the standard, not `connection: close`, so I highly recommend the removal.

**Comment by bitinn** [comment]
I think it's sane to go with the node.js `http.request` default, when you use an agent we could default to `keep-alive` (aka not set the connection close header).

https://nodejs.org/api/http.html#http_http_request_options_callback

PS: on fetch spec `connection` is forbidden header so I think end-user have no control there.

**Comment by dendril** [comment]
Thanks a lot @bitinn. When do you plan to remove the code and tag a new version?

**Comment by jimmywarting** [comment]
This just became interesting to me

**Comment by dendril** [comment]
@bitinn ping

**Comment by bitinn** [comment]
@dendril this is fixed in master, but I am holding off from a new release for now.

note that we don't handle anything about connection pool, it is not `node-fetch` responsibility to close socket etc.

**Comment by dendril** [comment]
cool, I'll keep monitoring for the next release.

Thanks a lot for your time!

**Comment by bitinn** [comment]
this is now released as v1.5.2
